Tests/SwiftCLITests/Fixtures.swift

```diff
@@ -114,60 +117,78 @@ class EmptyCmd: Command {
 }
 
 class Req1Cmd: EmptyCmd {
-    let req1 = Parameter(completion: .function("_ice_targets"))
+    @Param.Required(completion: .function("_ice_targets"))
+    var req1: String
 }
 
 class Opt1Cmd: EmptyCmd {
-    let opt1 = OptionalParameter()
+    @Param.Optional
+    var opt1: String?
 }
 
 class Req2Cmd: EmptyCmd {
-    let req1 = Parameter(completion: .filename)
-    let req2 = Parameter(completion: .values([
+    @Param.Required(completion: .filename)
+    var req1: String
+    
+    @Param.Required(completion: .values([
         ("executable", "generates a project for a cli executable"),
         ("library", "generates project for a dynamic library"),
         ("other", "")
     ]))
+    var req2: String
 }
 
 class Opt2Cmd: EmptyCmd {
-    let opt1 = OptionalParameter(completion: .filename)
-    let opt2 = OptionalParameter(completion: .values([
+    @Param.Optional(completion: .filename)
+    var opt1: String?
+    
+    @Param.Optional(completion: .values([
         ("executable", "generates a project for a cli executable"),
         ("library", "generates project for a dynamic library")
     ]))
+    var opt2: String?
 }
 
 class Opt2InhCmd: Opt2Cmd {
-    let opt3 = OptionalParameter()
+    @Param.Optional var opt3: String?
```

Comment:
Do you need `Param.Optional` and `Param.Required`, or can it just be inferred if the property is optional or not?

Reply:
I tried to have it be inferred but was unable to get it to work. The only way I could think to do that was have `Optional` conditionally conform to `ConvertibleFromString` when `Wrapped` does, but the compiler crashes when I add that conformance